addons/pos_self_order/models/product_product.py

```diff
@@ -49,6 +50,11 @@ def _load_pos_self_data_fields(self, config_id):
         params = super()._load_pos_self_data_fields(config_id)
         params += ['description_self_order']
         return params
+    
+    @api.model
+    def _load_pos_self_data_domain(self, data):
+        domain = super()._load_pos_self_data_domain(data)
+        return AND([domain, [('self_order_available', '=', True)]])
```

Comment:
It seems nice but on the runbot, when clicking on this, it generates a traceback on the self screen. Can you check if it is linked to your PR ?
![image](https://github.com/user-attachments/assets/e7d59a85-fbda-4ba0-82a6-4d5e3fe5978c)

Reply:
I rebased it and it has been solved.